siriuspy/siriuspy/devices/intlkctrl.py

```diff
@@ -149,86 +149,136 @@ def __init__(self, devname=None, props2init='all', **kwargs):
             devname = self.DEVICES.CAX
         if devname not in self.DEVICES.ALL:
             raise NotImplementedError(devname)
+        self._error_log = ''
         super().__init__(devname, props2init=props2init, **kwargs)
 
+    @property
+    def error_log(self):
+        """Return error log for last check method invoked."""
+        return self._error_log
+
     @property
     def is_hutchA_intlk_search_done(self):
         """."""
-        return self['A:PPS01:SEARCH_OK'] == 1
+        status_ok = self['A:PPS01:SEARCH_OK'] == 1
+        if not status_ok:
+            self._error_log = 'hutch A interlock search not Ok.'
```

Comment:
I think error log should return to empty string if command is successful

Reply:
Contrary to what I talked to you offline moments agor, ana, I thought about it and I think now that it is more useful for `error_log` to record not the current error state of the BL but the last error observed during the process of opening the BL, regardless of when/what point this happened. For this we should not return empty string in `is_*` methods when they succeed but only when `cmd_beamline_open` specifically succeeds.

Reply:
yes. done.